src/main/java/com/snowflake/kafka/connector/internal/streaming/TopicPartitionChannel.java

```diff
@@ -129,19 +135,28 @@ public void insertRecordToBuffer(SinkRecord record) {
     }
   }
 
-  /* TODO: SNOW-529753 Deal with rebalances and restarts */
-  /* TODO: SNOW-536429 For EOS, also fetch the offset from snowflake during rebalance/restart */
-  private void init(long recordOffset) {
+  /* For EOS, fetch the offset from snowflake during only during first time after connector starts and record is received in Kafka Connector */
+  private void fetchLastCommittedOffsetToken() {
+    String offsetToken = this.channel.getLatestCommittedOffsetToken();
     try {
-      startCleaner(recordOffset);
-    } catch (Exception e) {
-      LOGGER.error("Cleaner and Flusher threads shut down before initialization");
+      if (offsetToken == null) {
+        this.offsetPersistedInSnowflake.set(-1);
```

Comment:
Should we pull `-1L` as a constant somewhere and treat that as "NO_OFFSET_TOKEN" here and elsewhere?

Reply:
Will do. "no magic constants"